server/service/src/programs/encounter/encounter_updated.rs

```diff
@@ -31,7 +31,7 @@ pub(crate) fn update_encounter_row(
         .pop();
     let id = match row {
         Some(row) => row.0.id,
-        None => uuid(),
+        None => sha256(&doc.name),
```

Comment:
I added a comment on why sha256 is used. An alternative, would be to use the name as the ID but having an ID is nicer to put, for example, into an web URL. I could also have used an ID for the encounter/contact_trace name. I didn't do this because the program_enrolment uses a predefined pattern to avoid duplicated enrolments. While a human readable document name is nicer to debug I wonder if I should just use an ID for patient, encounter and contact tracing documents...

Reply:
My main comment wasn't about sha256, it was mainly about doc.name being used for id (that required a more thinking to validate, so I am trying to figure out the benefits of it). 
So is the only real benefit (if sha256 is removed), you can search for all encounters or enrolement or contact traces for a patient by searching for ID containing patient id ?

Reply:
The idea is that a document is similar to a file. Simplified, it stores some data and has a path. Using this analogy, all documents of a single patient are put in a single folder by starting the document name with the patient id. For example, you could query all patient documents in one query using a startsWith query. However, this is currently not done since there is a patient_id on the document now. Main usage is for program enrolments to avoid duplicates (so to say, by storing the enrolment in a predefined file in the patient's dir)